internal/controller/reconciler.go

```diff
@@ -30,21 +30,21 @@ type Config struct {
 	NamespacedNameFilter NamespacedNameFilterFunc
 }
 
-// Implementation is a reconciler for Kubernetes resources.
+// Reconciler reconciles Kubernetes resources of a specific type.
 // It implements the reconcile.Reconciler interface.
 // A successful reconciliation of a resource has the two possible outcomes:
 // (1) If the resource is deleted, the Implementation will send a DeleteEvent to the event channel.
 // (2) If the resource is upserted (created or updated), the Implementation will send an UpsertEvent
 // to the event channel.
-type Implementation struct {
-	cfg Config
+type Reconciler struct {
```

Comment:
This type could have been private.  The controller.Register function is enough to expose controller-registration functionality.

However, because of the way we generate fakes for our unit tests, we need this type to be public.

Explanation:
The generated fakes are in a separate package and reference the package where the interface was declared. For example:
```
package controllerfakes
...
var _ controller.Getter = new(FakeGetter)
```

This means we cannot have the unit test for the Reconciler in package `controller` - that would lead to a cycling dependency. Instead, we have the unit test in `controller_test`, which requires Reconciler to be public.